### PR TITLE
Parse lit tag

### LINKED ISSUE #3848: Support `lit` OSM key
For pedestrian safety.

Easy:
- parse boolean from OSM into the graph
- add one bit on directed edge for lit

Medium-easy:
- decide how to use it in EdgeCost of pedestrian, e.g.
  -  `use_lit` costing option and use it directly?
  - `safe` costing option and wrap the `lit` attribute with others such as road class penalties?
  - other ideas?

### PR DESCRIPTION
**Pull request by chrstnbwnkl**
fixes #3848 

This PR adds parsing of the `lit` tag:

  - parse the possible values and map them to a boolean value in `graph.lua` 
  - add `lit` as a member variable to `OSMWay`
  - add `lit` as a member variable to the `DirectedEdge` class, thereby allocating one of the remaining bits
  - add the `lit` tag and the values we parse to `taginfo.json`
  
I also added a short parameterized test going through the different possible values of `lit`, including its absence.

### CODE REVIEW

File: lua/graph.lua
Comment (chrstnbwnkl) [inline lua/graph.lua:416]: `lit=limited` is a special case that, according to the OSM wiki mainly occurs in Austria and Germany, where it signifies that a street light marked with a ["Laternenring"](https://de.wikipedia.org/wiki/Laternenring) is not turned on during the whole night, so I guess we cannot reliably set the marked edge as `lit=true`

**Comment by nilsnolde** [comment]
Yay thanks! Got my approval, but someone else should review and approve:)

**Review by kevinkreiser — CHANGES_REQUESTED** [review]
so other than the change to osmway.h this looks good, we'll ship it after we get that fixed up. thanks!

**Review by kevinkreiser — APPROVED** [review]

**Pull request merged**
